### PR TITLE
feat(replay): add before/after labels and tooltips to slider diff

### LINKED ISSUE #74458: [Hydration Errors] Improve UI labels inside the slider-diff-tool
Add "Before" and "After" labels to the top left & right corders of the slider-diff tool. This should help people understand it slightly better.

Image

### PR DESCRIPTION
**Pull request by aliu39**
Fixes https://github.com/getsentry/sentry/issues/74458
The Before tooltip has been changed to "How the initial server-rendered page looked."
Screenshot 2024-07-19 at 12 05 55 PM
Screenshot 2024-07-19 at 12 06 07 PM

### CODE REVIEW

**Review by ryan953 — APPROVED** [review]

File: static/app/components/replays/diff/replaySliderDiff.tsx [hunk 1]
```diff
@@ -23,21 +25,35 @@ export function ReplaySliderDiff({leftOffsetMs, replay, rightOffsetMs}: Props) {
...
-            width={width}
-          />
-        ) : (
-          <div />
-        )}
-      </Positioned>
-    </WithPadding>
+    <Fragment>
+      <Header>
+        <Tooltip title={t('How the initial server-rendered page looked like.')}>
```
Comment (michellewzhang) [inline static/app/components/replays/diff/replaySliderDiff.tsx:30]:
```suggestion
        <Tooltip title={t('What the initial server-rendered page looked like.')}>
```
```suggestion
        <Tooltip title={t('How the initial server-rendered page looked.')}>
```
THIS IS NITTY i feel like one of these flows better

**Review by michellewzhang — APPROVED** [review]

File: static/app/components/replays/diff/replaySliderDiff.tsx (hunk 1 above)
Comment (c298lee) [reply static/app/components/replays/diff/replaySliderDiff.tsx:30]: I vote for the second one ('How the initial server-rendered page looked.')

**Comment by aliu39** [comment]
@Jesse-Box let us know what you think! Also just realized, we could use a question mark tooltip to make the tooltip more discoverable. Any thoughts on this?

**Comment by michellewzhang** [comment]
> @Jesse-Box let us know what you think! Also just realized, we could use a question mark tooltip to make the tooltip more discoverable. Any thoughts on this y'all?

i'm a fan of the question mark

**Review by Jesse-Box — APPROVED** [review]

**Pull request merged**